docs/guides/migrate/migration-guide.md

```diff
@@ -113,8 +113,8 @@ object MainApp extends ZIOAppDefault {
     - }
     + def foreachParN[A](n0: => Int)(a0: => Iterable[A]) = 
     +   ZIO.suspendSucceed {
-    +    val n = n0 
-    +    val a = a0
+    +    lazy val n = n0 
```

Comment:
The whole `suspendSucceed` code block is lazy, so this change won't make the code more lazy unless `n` and `a` are not used within the body of the function.

Reply:
Just that it's what it says the paragraph above the code that we must use `lazy val`s:

> All parameters to operators returning an effect [should be by-name](https://zio.dev/next/howto/migrate/zio-2.x-migration-guide/#lazy-evaluation-of-parameters). Also, we should be sure to capture any parameters that are referenced more than once as a lazy val in our implementation to prevent double evaluation.

Reply:
I've updated the commit changing the paragraph and reverting the code to vals.